packages/android_intent/android/src/test/java/dev/flutter/plugins/androidintent/MethodCallHandlerImplTest.java

```diff
@@ -0,0 +1,137 @@
+package dev.flutter.plugins.androidintent;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.robolectric.Shadows.shadowOf;
+
+import android.app.Application;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import androidx.annotation.Nullable;
+import androidx.test.core.app.ApplicationProvider;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel.Result;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+@RunWith(RobolectricTestRunner.class)
+public class MethodCallHandlerImplTest {
+  private Context context;
+  private IntentSender sender;
+  private MethodCallHandlerImpl methodCallHandler;
+
+  public MethodCallHandlerImplTest() {}
+
+  @Before
+  public void setUp() {
+    context = ApplicationProvider.getApplicationContext();
+    sender = new IntentSender(null, null);
+    methodCallHandler = new MethodCallHandlerImpl(sender);
+  }
+
+  @Test
+  public void onMethodCall_doesNothingWhenContextIsNull() {
+    sendLaunchMethodCall("foo", null, null, null, null);
```

Comment:
Opinion: When the details that are relevant to a particular test case are hidden in other methods, it can make it more difficult for developers to understand what is relevant under test vs what is orthogonal to the behavior under test.